.github/workflows/ci.yaml

```
@@ -49,11 +41,49 @@ jobs:
     - name: Clippy
-      run: cargo clippy -- -Dwarnings
\ No newline at end of file
+      run: cargo clippy -- -Dwarnings
+
+  # The capnp-generated code is committed rather than built from build.rs, so it
+  # can drift from the schema if someone edits the .capnp file without
+  # regenerating. Regenerate here and fail if the committed code is stale.
+  check_capnp_generated:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
```

actionlint (1.7.12)

[error] the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)